tests/SIL.LCModel.Core.Tests/App.config

```diff
@@ -8,16 +8,8 @@
 			</listeners>
 		</trace>
 	</system.diagnostics>
-  <runtime>
-    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
-      <dependentAssembly>
-        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-12.0.0.0" newVersion="12.0.0.0" />
-      </dependentAssembly>
-      <dependentAssembly>
-        <assemblyIdentity name="icu.net" publicKeyToken="416fdd914afa6b66" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-2.8.0.0" newVersion="2.8.0.0" />
-      </dependentAssembly>
-    </assemblyBinding>
-  </runtime>
+  <!-- These two dllmap entries are copied from icu.net.dll.config. They are needed to get tests to pass on Ubuntu 22.04. -->
+  <!-- See https://github.com/sillsdev/icu-dotnet/issues/181 and https://github.com/sillsdev/TheCombine/issues/1768       -->
+  <dllmap os="!windows,osx" dll="libdl.so" target="libdl.so.2" />
```

Comment:
Having `artifacts/Release/net461/icu.net.dll.config` present [didn't help](https://github.com/sillsdev/liblcm/actions/runs/3687538562/jobs/6241242366), either, so I tried this. I thought the config file that mattered was for the entry assembly, not the dependency.

Reply:
ok, I figured it out. Wasn't caused by icu.net but by two problems in liblcm code:

1. tests didn't find our custom libicuuc shared lib because it lives in a non-standard location (`/usr/lib/fieldworks/lib/x64`). This can be solved by sourcing `environ` before running the tests.
2. The SpellChecking code also references `libdl.so`, so `SIL.LCModel.Core.dll.config` also needs the dll mappings.

I added two commits with these changes.

Reply:
Thanks, @ermshiperete!